include/nbl/builtin/hlsl/subgroup/arithmetic_portability_impl.hlsl

```diff
@@ -0,0 +1,435 @@
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+
+uint threadIndex : SV_GroupIndex; // Use HLSL terminology or follow GLSL (i.e. call them invocations instead of threads)?
```

Comment:
follow GLSL, also dont define global variables like this

Reply:
From my tests, HLSL doesn't allow defining SV_* variables inside a function scope. Do you have a suggestion?

Reply:
IIRC you can just `#define gl_ SV_`

Reply:
> IIRC you can just `#define gl_ SV_`

What I mean is, in HLSL you can't do something like
```
void myFun() {
  uint threadIdx: SV_GroupIndex;
}
```
Instead it has to be in the form of a parameter or global, or part of a struct.
For example:
```
void myFun(uint threadIdx: SV_GroupIndex) {
  ...
}
```

Reply:
then do 
```hlsl
uint gl_LocalInvocationIndex : VS_GroupIndex;
```
or this stuff
https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst#builtin-variables